src/ptvsd/adapter/components.py

```diff
@@ -34,19 +34,29 @@ class Component(util.Observable):
     to wait_for() a change caused by another component.
     """
 
-    def __init__(self, session, stream):
+    def __init__(self, session, stream=None, channel=None):
+        assert (stream is None) ^ (channel is None)
```

Comment:
>(stream is None) ^ (channel is None) [](start = 15, length = 36)

The rare XOR operator usage :)

Reply:
While we're at it, I feel very stymied by the lack of logical implication operator in Python. 😂